security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -957,7 +958,25 @@ private HttpRequestBase makeRequest(HttpServletRequest request, String sURL) thr
         try {
             URL url = new URL(sURL);
             URI uri = buildUri(url);
-            HttpMethod meth = HttpMethod.resolve(request.getMethod());
+            String method = request.getMethod();
+            // handles webdav specific verbs
+            if (PropFindMethod.METHOD_NAME.equalsIgnoreCase(method)) {
+                PropFindMethod pfm = new PropFindMethod(uri);
+                int contentLength = request.getContentLength();
+                ServletInputStream inputStream = request.getInputStream();
+                HttpEntity entity = new InputStreamEntity(inputStream, contentLength);
+                pfm.setEntity(entity);
+                return pfm;
+            } else if (SearchMethod.METHOD_NAME.equalsIgnoreCase(method)) {
+                SearchMethod sm = new SearchMethod(uri);
+                int contentLength = request.getContentLength();
+                ServletInputStream inputStream = request.getInputStream();
+                HttpEntity entity = new InputStreamEntity(inputStream, contentLength);
+                sm.setEntity(entity);
+                return sm;
+            }
```

Comment:
I'm wondering in which case the two conditions couldn't be fullfiled?

Reply:
if the coming request is a completely custom HTTP verb ? But then I'd expect an exception to be thrown here:
https://github.com/georchestra/georchestra/pull/3483/files#diff-557c2eb148c461ecfcde83d38082b5270605be793cd92a707660c2f38e527adfR1053

Reply:
do you think it deserves an extra test ?

Reply:
trying an extra test shows that `switch(null)` throws an npe.

Reply:
I added a test & a guard to avoid the NPE.